lib/PlansConfigurator/SourcesLayout.tsx

```diff
@@ -0,0 +1,45 @@
+import { useContext } from "react";
+import { ISourcesLayout } from "./interface";
+import { GlobalContext } from "../context";
+
+export default function SourcesLayout({ sourceNames }: ISourcesLayout) {
+  const { state, setState } = useContext(GlobalContext);
+
+  const handleSelectSource = (source: string) => {
+    setState({ selectedSource: source });
+  };
+  
+  const sourceList = Object.entries(sourceNames);
+
+  return (
+    <div className="mobbex-pc-sources-layout">
+      {sourceList.map(([sourceReference, sourceName]) => {
+        const isSelected = state.selectedSource === sourceReference;
+        return (
+          <button
+            type="button"
+            key={sourceReference}
+            className={`mobbex-pc-payment-methods-sources ${
+              isSelected ? "mobbex-source-selected" : ""
+            }`}
+            onClick={() => handleSelectSource(sourceReference)}
```

Comment:
Elimino la función `handleSelectSource`, porque también se puede hacer un una línea con el `setState` directamente

```suggestion  
  const sourceList = Object.entries(sourceNames);

  return (
    <div className="mobbex-pc-sources-layout">
      {sourceList.map(([sourceReference, sourceName]) => {
        const isSelected = state.selectedSource === sourceReference;
        return (
          <button
            type="button"
            key={sourceReference}
            className={`mobbex-pc-payment-methods-sources ${
              isSelected ? "mobbex-source-selected" : ""
            }`}
            onClick={() => setState({ selectedSource: sourceReference })}
```